Sources/SwiftSyntax/SyntaxNodes.swift.gyb.template

```diff
@@ -100,11 +100,9 @@ public struct ${node.name}: ${base_type}Protocol, SyntaxHashable {
 %
   public let _syntaxNode: Syntax
 
-  /// Converts the given `Syntax` node to a `${node.name}` if possible. Returns
-  /// `nil` if the conversion is not possible.
-  public init?(_ syntax: Syntax) {
-    guard syntax.raw.kind == .${node.swift_syntax_kind} else { return nil }
-    self._syntaxNode = syntax
+  public init?<S: SyntaxProtocol>(_ syntaxNode: S) {
```

Comment:
`SyntaxBaseNodes.swift.gyb` has `init<S: ${node.name}Protocol>(_ syntax: S)`. The parameter name should be consistent. Same for `init?<S: SyntaxProtocol>(_:)` in `SyntaxBaseNodes.swift.gyb`

Reply:
Which do you prefer?